### ISSUE TITLE
Server.go : correct order to close all the different connections properly

### ISSUE THREAD

**Issue by TristerosEmpire** [simplesteph/grpc-go-course#9]
With the current code, if we handle the errors at the connection closing with MongoDB and at the listener closing, we've got errors :` "accept tcp [::]:50051: use of closed network connection"`.

So we need to, first, close the connection with MongoDB, then, close the listener and, finally, stop the server.

Solution :

```go
// Correct order :
// First we close the connection with MongDB:
fmt.Println("Closing MongoDB Connection")
client.Disconnect(context.TODO())
if err := client.Disconnect(context.TODO()); err != nil {
	log.Fatalf("Error on disconnection with MongoDB : %v", err)
}
// Second step : closing the listener
fmt.Println("Closing the listener")
if err := lis.Close(); err != nil {
    log.Fatalf("Error on closing the listener : %v", err)
}
// Finally, we close the server
fmt.Println("Stopping the server")
s.Stop()
fmt.Println("End of Program")`
```

**Comment by simplesteph** [comment]
Thanks, makes sense. Feel free to propose a PR

**Comment by simplesteph** [comment]
Fixed by #10
